src/flora/communicator/grpc_communicator.py

```diff
@@ -57,12 +56,17 @@ def __init__(
         # configurable gRPC buffer sizes
         self.max_send_message_length = max_send_message_length
         self.max_receive_message_length = max_receive_message_length
+        # configurable retry behavior
+        self.retry_delay = retry_delay
+        self.max_retries = max_retries
 
         # ---
-        self.server: Optional[grpc.Server] = None
-        self.client: Optional[GrpcClient] = None
+        self.server = None
+        self.client = None
```

Comment:
should maybe make these @property (getter) & setter to pack error-handling in there